doc/design/doi-generation-2.md

```diff
@@ -0,0 +1,229 @@
+Every public published version of a Dandiset will receive a **Version DOI**.
+
+For example:
+Dandiset DOI: `https://doi.org/10.48324/dandi.000027/`
+Version DOI: `https://doi.org/10.48324/dandi.000027/0.210831.2033`
+
+The Dandiset DOI will always refer to the DLP
+
+At creation the `Dandiset DOI` will be a DataCite `Draft DOI`, but will be "promoted" to a DataCite `Findable DOI` as soon as there is a published version.
+A `Version DOI` will be created as a `Findable DOI`.
+
+## The current approach
+
+- [initial design doc](./doi-generation-1.md)
+- overall:
+   - leave DOI absent upon dandiset creation
+   - upon publication
+      - inject fake DOI (but do not save) and validate
+      - after validation, create a new `Version DOI` (function `create_doi`)
+      - publish dandiset
+
+### Issues with the Existing Approach
+
+- [Stop injecting "fake" DOIs into draft dandisets](https://github.com/dandi/dandi-archive/issues/1709)
+- [Unpublished Dandisets display a DOI under `Cite As`](https://github.com/dandi/dandi-archive/issues/1932)
+
+## Background
+
+Initially proposed/discussed in
+
+- [Create and maintain a "Findable" DOI for the Dandiset as a whole](https://github.com/dandi/dandi-archive/issues/1319)
+
+and boils down to the adoption of approach of Zenodo of having a DOI which always points to the latest version of the record.
+[Zenodo uses the language](https://support.zenodo.org/help/en-gb/1-upload-deposit/97-what-is-doi-versioning) `Concept DOI` to mean a top-level DOI that references all versions, which we will refer to as `Dandiset DOI`.
+
+DataCite allows for three types of DOIs ([DataCite](https://support.datacite.org/docs/what-does-the-state-of-the-doi-mean)):
+
+- `Draft`. We do not use those.
+  *Can be deleted, and they require only the identifier itself in order to be created or saved. They can be updated to either Registered or Findable DOIs. Registered and Findable DOIs may not be returned to the Draft state, which means that changing the state of a Draft is final.*
+- `Registered`. Like `Findable` but not indexed for search, so we do not use them.
+- `Findable`. Is the type we use for published dandisets.
+  Requires to be valid (pass validation to fit the datacite schema) to be created.
+
+## Proposed Solution
+
+- For **Public dandisets**:
+    - Upon creation:
+        - mint a `Dandiset DOI` (a DataCite `Draft DOI`) `10.48324/dandi.{dandiset.id}` with *minimal metadata* entered during creation request (title, description, license)
+        - URL should point be DLP `https://dandiarchive.org/dandiset/{dandiset.id}`
+        - If minting a DOI fails, we need to raise exception to inform developers about the issue but proceed with the creation of the dandiset.
+    - Upon updates to a draft dandiset metadata **prior to first publication**:
+        - Update the datacite metadata of the `Draft DOI`, (leave as draft)
```

Comment:
```suggestion
        - Update the datacite metadata and if still a `Draft` DOI, try to make it a `Findable` DOI
           - if `Findable` fails, try to update metadata as a `Draft` DOI
```

Reply:
this was to provide citeable DOIs for Draft Dandisets with zarrs... for now let's postpone this and if needed, will implement/enable later.